anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+            self.files_dir, 'matching_taxonomy.tsv')
+        self.pathpickle_dico_taxo = os.path.join(
+            self.outputdirectory, 'dico_taxo_code_species.pickle')
+
+        if not os.path.exists(pathpickle_dico_taxo):
+            self.dicolevel = self.make_dicolevel(taxofiles)
+            with open(pathpickle_dico_taxo, 'wb') as handle:
+                pickle.dump(dicolevel, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    else:
+        with open(pathpickle_dico_taxo, 'rb') as handle:
+            dicolevel = pickle.load(handle)
+
+
+    dico_low_ident = {}
+    for genes in genesfiles:
+        dico_low_ident_genes={}
+        pathfa = os.path.join(genesfilesdir, genes)
+        dico_low_ident_genes=self.self.creatsubfa_ident(dicolevel, pathfa, outputdirectory, genes,dico_low_ident_genes)
+        pathpickle_dico_ident = os.path.join(
+            outputdirectory, genes+'_dico_low_ident.pickle')
+        with open(pathpickle_dico_ident, 'wb') as handle:
+            pickle.dump(dico_low_ident_genes, handle, protocol=pickle.HIGHEST_PROTOCOL)
+        if genes not in dico_low_ident:
+            dico_low_ident[genes]=dico_low_ident_genes
+        else:
+            dico_low_ident[genes]=dico_low_ident[genes].update(dico_low_ident_genes)
+
+    pathpickle_dico_ident = os.path.join(
+        outputdirectory, 'dico_low_ident.pickle')
+    with open(pathpickle_dico_ident, 'wb') as handle:
+        pickle.dump(dico_low_ident, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    print(dico_low_ident)
+
+
+    def make_dicolevel(self,taxofiles):
+        with open(taxofiles, 'r') as taxotsv:
+            linestaxo = taxotsv.readlines()
+            dicolevel = {}
+            for line in linestaxo:
+                names = line.split('\t')
+                code = names[0]
+                leveltaxos = names[1].split(';')
+                for leveltaxo in leveltaxos:
+                    leveltaxo=leveltaxo.rstrip()
+                    if leveltaxo not in dicolevel:
+                        dicolevel[leveltaxo] = [code]
+                        continue
+                    if code in dicolevel[leveltaxo]:
+                        continue
+                    else:
+                        dicolevel[leveltaxo] = dicolevel[leveltaxo] + [code]
+        return(dicolevel)
+
+
+
+    def match_ident(self, fasta, codes, listindex, dicolevel=False):
+        for code in codes:
+            if code in fasta.ids:
+                index = fasta.ids.index(code)
+                if index:
+                    listindex.append(index)
+        return(listindex)
+
+
+
+
+
+
+
+    def multidiamond(self, listeprocess,outputdirectory,dico_low_ident):
+        manager = multiprocessing.Manager()
+        input_queue = manager.Queue()
+        output_queue = manager.Queue()
+
+        num_listeprocess = len(listeprocess)
+
+        progress.new('Aligning amino acid sequences for genes in gene clusters', progress_total_items=num_listeprocess)
```

Comment:
These progress messages makes zero sense. You are not aligning amino acid sequences here for gene clusters here. Please pay attention to these kinds of details.

Reply:
Your code is doing a lot of things without letting anyone know what is really going on in general. You should use `progress.new/update/end` to make sure the user has an idea what is the code doing.